arangod/IResearch/IResearchAqlAnalyzer.cpp

```diff
@@ -652,11 +722,28 @@ bool AqlAnalyzer::reset(irs::string_ref const& field) noexcept {
       optimizer.createPlans(std::move(plan), _query->queryOptions(), false);
 
       _plan = optimizer.stealBest();
+
+      // try to optimize
+      if(optimize(field)) {
+        _optimizedField.assign(field.c_str(), field.size());
+        return true;
+      }
+
     } else {
+      // plan exists
+
```

Comment:
If we are in "optimized" mode - why wasting time on node values? We could check here _preCalculated value.